plugins/kotlin/uast/uast-kotlin-base/src/org/jetbrains/uast/kotlin/declarations/AbstractKotlinUVariable.kt

```diff
@@ -71,22 +70,19 @@ abstract class AbstractKotlinUVariable(
         }
 
     private fun AbstractKotlinUVariable.buildAnnotations(): List<UAnnotation> {
-        val sourcePsi = sourcePsi ?: return psi.annotations.map { WrappedUAnnotation(it, this) }
+        val sourcePsi = sourcePsi ?: return javaPsi.annotations.map { WrappedUAnnotation(it, this) }
         val annotations = SmartList<UAnnotation>()
         val hasInheritedGenericType = baseResolveProviderService.hasInheritedGenericType(sourcePsi)
         if (!hasInheritedGenericType) {
             annotations.add(KotlinNullabilityUAnnotation(baseResolveProviderService, sourcePsi, this))
         }
-        if (sourcePsi is KtModifierListOwner) {
-            sourcePsi.annotationEntries
-                .filter { acceptsAnnotationTarget(it.useSiteTarget?.getAnnotationUseSiteTarget()) }
-                .mapTo(annotations) { baseResolveProviderService.baseKotlinConverter.convertAnnotation(it, this) }
-        }
+        // NB: we can't use sourcePsi.annotationEntries directly due to annotation use-site targets.
+        baseResolveProviderService.getPsiAnnotations(javaPsi).asSequence()
+            .mapNotNull { (it as? KtLightElement<*, *>)?.kotlinOrigin as? KtAnnotationEntry }
+            .mapTo(annotations) { baseResolveProviderService.baseKotlinConverter.convertAnnotation(it, this) }
```

Comment:
Now I better understand why we did this: https://github.com/JetBrains/intellij-community/blob/master/plugins/kotlin/uast/uast-kotlin-base/src/org/jetbrains/uast/kotlin/declarations/KotlinUMethod.kt#L74-L80